cloudinit/sources/DataSourceAkamai.py

```diff
@@ -0,0 +1,372 @@
+            return MetadataAvailabilityResult.defer
+
+        return MetadataAvailabilityResult.available
+
+    def _get_network_context_managers(
+        self,
+    ) -> List[Tuple[Union[Any, EphemeralIPNetwork], bool]]:
+        """
+        Returns a list of context managers which should be tried when setting
+        up a network context.  If we're running in init mode, this return a
+        noop since networking should already be configured.
+        """
+        network_context_managers: List[
+            Tuple[Union[Any, EphemeralIPNetwork], bool]
+        ] = []
+        if self.local_stage:
+            # at this stage, networking isn't up yet.  To support that, we need
+            # an ephemeral network
+
+            # find the first interface that isn't lo or a vlan interface
+            interfaces = get_interfaces_by_mac()
+            interface = None
+            preferred_prefixes = self.ds_cfg["preferred_mac_prefixes"]
+            for mac, inf in interfaces.items():
+                # try to match on the preferred mac prefixes
+                if any(
+                    [mac.startswith(prefix) for prefix in preferred_prefixes]
+                ):
+                    interface = inf
+                    break
+
+            if interface is None:
+                LOG.warning(
+                    "Failed to find default interface, attempting DHCP on "
+                    "fallback interface"
+                )
+                interface = find_fallback_nic()
+
+            network_context_managers = []
+
+            if self.ds_cfg["allow_ipv6"]:
+                network_context_managers.append(
+                    (
+                        EphemeralIPNetwork(
+                            self.distro,
+                            interface,
+                            ipv4=False,
+                            ipv6=True,
+                        ),
+                        True,
+                    ),
+                )
+
+            if self.ds_cfg["allow_ipv4"] and self.ds_cfg["allow_dhcp"]:
+                network_context_managers.append(
+                    (
+                        EphemeralIPNetwork(
+                            self.distro,
+                            interface,
+                            ipv4=True,
+                        ),
+                        False,
+                    )
+                )
+        else:
+            if self.ds_cfg["allow_ipv6"]:
+                network_context_managers.append(
+                    (
+                        noop(),
+                        True,
+                    ),
+                )
+
+            if self.ds_cfg["allow_ipv4"]:
+                network_context_managers.append(
+                    (
+                        noop(),
+                        False,
+                    ),
+                )
+
+        return network_context_managers
+
+    def _make_requests(self, use_v6: bool = False) -> bool:
```

Comment:
Nit: This method not only make requests, it does change the state of the ds, by setting `self.metadata` and `self.userdata_raw`. Could think a better name?